src/GitHub.App/Services/RepositoryPublishService.cs

```diff
@@ -49,5 +52,11 @@ public string LocalRepositoryName
                                  return repo.RemoteRepo;
                              });
         }
+
+        async Task<IRepository> GetActiveRepo()
+        {
+            await ThreadingHelper.SwitchToMainThreadAsync();
+            return vsGitServices.GetActiveRepo();
```

Comment:
I looked at the docs and the code and `GetService` should be totally safe to call now. Sooo we really need to test this again and see if we still get deadlocks.